ios/Classes/BetterPlayer.m

```diff
@@ -764,20 +785,27 @@ - (void)pictureInPictureControllerDidStartPictureInPicture:(AVPictureInPictureCo
 
 - (void)pictureInPictureControllerWillStopPictureInPicture:(AVPictureInPictureController *)pictureInPictureController  API_AVAILABLE(ios(9.0)){
     [self setIsPipMode:false];
+    [self hideBlackCoverView];
+    [self hideLimitedPlanCoverView];
```

Comment:
Sorry,
You added the `hideLimitedPlanCoverView` when callback `pictureInPictureControllerWillStopPictureInPicture`.

The function is called "Will" Stop PIP.
I'm afraid of showing premium video for a moment during PIP -> normal mode.
Is it ok?

Reply:
> Is it ok?

It's ok.
This "WillStopPictureInPicture" func is only called after it completely become normal mode. (While testing, the "DidStopPictureInPicture" never got called)

Beside of this, the "WillStartPictureInPicture" work perfectly as its name, that is why i have to create a `_isPipMode` variable, because the `_pipController.isPictureInPictureActive` didn't work on the transition moment.

Reply:
Thank you for describing. 👍 
I understand. Thanks.